core/src/main/resources/hudson/model/AbstractModelObject/editDescription.jelly

```diff
@@ -31,15 +31,16 @@ THE SOFTWARE.
     <st:include page="sidepanel.jelly" />
     <l:main-panel>
       <form action="submitDescription" method="post">
+        <j:set var="description" value="${it == app ? it.primaryView.description : it.description}" />
         <table>
           <f:entry help="${app.markupFormatter.helpUrl}">
             <!-- JENKINS-37360: Since Jenkins has a description (system message), prevent that from showing here.
              Instead, use primary view's, as Jenkins's cannot be edited anyway (see Jenkins#submitDescription) -->
-            <f:textarea name="description" value="${it == app ? it.primaryView.description : it.description}"
+            <f:textarea name="description" value="${description}"
                         codemirror-mode="${app.markupFormatter.codeMirrorMode}" codemirror-config="${app.markupFormatter.codeMirrorConfig}" previewEndpoint="/markupFormatter/previewDescription"/>
           </f:entry>
         </table>
-        <f:submit />
+        <f:submit value="${empty(description) ? '%Submit' : '%Save'}" />
```

Comment:
I don't think we need to differentiate, save would work all the time (and revert other related changes
```suggestion
        <f:submit value="${%Save}" />
```

Reply:
Sorry, my bad I had misunderstood. It's set to 'Save' now :)